src/compute-client/src/protocol/response.rs

```diff
@@ -570,33 +701,55 @@ impl Arbitrary for SubscribeResponse<mz_repr::Timestamp> {
     }
 }
 
+/// The contents of a subscribe batch.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum SubscribeBatchContents<T = mz_repr::Timestamp> {
+    /// Regular updates.
+    Updates(Vec<(T, Row, Diff)>),
+    /// Updates that have been stashed in persist batches.
+    Stashed(Box<StashedSubscribeBatch>),
+    /// Error message.
+    Error(String),
+}
+
 /// A batch of updates for the interval `[lower, upper)`.
-#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
 pub struct SubscribeBatch<T = mz_repr::Timestamp> {
     /// The lower frontier of the batch of updates.
     pub lower: Antichain<T>,
     /// The upper frontier of the batch of updates.
     pub upper: Antichain<T>,
     /// All updates greater than `lower` and not greater than `upper`.
-    ///
-    /// An `Err` variant can be used to indicate e.g. that the size of the updates exceeds internal limits.
-    pub updates: Result<Vec<(T, Row, Diff)>, String>,
+    pub updates: SubscribeBatchContents<T>,
```

Comment:
Could change this to `Result<SubscribeBatchContents<T>, String>` so you don't need to inline the error into the contents.